### PR TITLE
WIP: Packetizer and Serial Line IP

### PR DESCRIPTION
**Pull request by micolous**
This adds a new `Packetizer` construct, used for splitting a stream of bytes into packets (ie: MAC sublayer of the data link layer).  This is useful for bringing serial connections into Scapy.

I implemented RFC 1055 Serial Line IP and made it a core layer.  The protocol itself is very simple, and it's useful for implementing both very similar protocols (such as nRF Sniffer's serial interface) and vaguely similar protocols (such as HDLC/PPP on serial).

The SLIP implementation includes enough glue and examples that are enough to work with `slattach` (Linux) and `slip2tun` (OSX).

I also added some functionality to the existing HDLC implementation to use it, but getting enough of the "glue" to make `pppd` work is very complex (as there are multiple negotiation steps).

There's a couple of more things I want to split out of this PR, so this is a slight work-in-progress:

* To make experimenting with interfaces in userspace friendlier, I implemented a `ICMPEcho_am` `AnsweringMachine`.  I use this in the examples here, but it is otherwise independent from the rest of the code.

* `SuperSocket` is hard coded to a single value for `recv` and `raw_recv` default calls, which is the default Ethernet MTU.  This is problematic for the packets I'm working with, so I have added `default_recv_size` as a constructor param for this.

* Adds a `SuperSocket.am()` convenience method.

### CODE REVIEW

**Comment by micolous** [comment]
Ah, I looks like my other branch was better than this new one, and I merged one that had less progress than before.  Let me try again...

**Comment by gpotter2** [comment]
This PR looked pretty good :/

File: scapy/compat.py [hunk 1]
```diff
@@ -9,13 +9,22 @@
...
+import sys
 
 import scapy.modules.six as six
 
+
+# Abstract Base Classes - this makes Python < 3.4 act like >= 3.4
+if sys.version_info >= (3, 4):
+    ABC = abc.ABC
+else:
+    ABC = abc.ABCMeta('ABC', (), {})
```
Comment (gpotter2) [inline scapy/compat.py]: Could you then update `scapy/contrib/http2.py` to use this ?
Also, I'm guessing you took that from the same stackoverflow answer than I did :smile: feel free to link to that if that's the case
